File: be-1-nestjs/src/auth/util.api-tokens.ts
import { Request, Response } from 'express';

export const getRequestTokens = (request: Request): { accessToken: string, refreshToken: string } => {
  const accessToken0 = request.headers.authorization?.split(' ')[1];
  const accessToken1 = request.headers['x-access-token'] as string;
  const refreshToken1 = request.headers['x-refresh-token'] as string;
  const accessToken = accessToken0 || accessToken1
  const refreshToken = refreshToken1
  return { accessToken, refreshToken }
}

export interface SetResponseTokensProps {
  accessToken?: string;
  refreshToken?: string;
}

export const setResponseTokens = (
  response: Response,
  { accessToken, refreshToken }: SetResponseTokensProps
) => {
  accessToken && response.setHeader('x-access-token', accessToken);
  refreshToken && response.setHeader('x-refresh-token', refreshToken);
}

File: be-2-flask/app/auth/util_api_tokens.py
from flask import request


def get_request_tokes() -> (str, str):
    """
    return (access_token, refresh_token)
    """
    auth_header = request.headers.get("Authorization", "")
    if auth_header.startswith("Bearer "):
        auth_header = auth_header[7:]
    access_token = auth_header or request.headers.get("x-access-token")
    refresh_token = request.headers.get("x-refresh-token")
    return access_token, refresh_token


def set_response_tokens(headers, access_token: str = None, refresh_token: str = None):
    if access_token != None:
        headers["x-access-token"] = access_token
    if refresh_token != None:
        headers["x-refresh-token"] = refresh_token
    return headers
